src/rp2_common/hardware_flash/include/hardware/flash.h

```diff
@@ -44,10 +44,7 @@
 
 #define FLASH_UNIQUE_ID_SIZE_BYTES 8
 
-// PICO_CONFIG: PICO_FLASH_SIZE_BYTES, size of primary flash in bytes, type=int, default=-1, group=hardware_flash
-#ifndef PICO_FLASH_SIZE_BYTES
-#warning PICO_FLASH_SIZE_BYTES is not set
-#endif
+// PICO_CONFIG: PICO_FLASH_SIZE_BYTES, size of primary flash in bytes, type=int, default=undefined, group=hardware_flash
```

Comment:
Why remove the warning? Are we expecting RP2040 boards to appear with no Flash fitted at all?

Reply:
I removed it in favor of a warning where it is used